### ISSUE TITLE
Thread-safe variant

### ISSUE THREAD

**Issue by piegamesde** [sunrise-choir/async-ringbuffer#2]
It'd be cool if there was a variant of this ring buffer that allowed reader and writer sides to be on different threads.

**Comment by vladimir-ea** [comment]
I agree, being able to read and write from 2 different async tasks would improve the usability of this crate.
